aws/data_source_aws_organizations_delegated_services.go

```diff
@@ -0,0 +1,86 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceAwsOrganizationsDelegatedServices() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceAwsOrganizationsDelegatedServicesRead,
+		Schema: map[string]*schema.Schema{
+			"account_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAwsAccountId,
+			},
+			"delegated_services": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"delegation_enabled_date": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"service_principal": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAwsOrganizationsDelegatedServicesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*AWSClient).organizationsconn
+
+	input := &organizations.ListDelegatedServicesForAccountInput{
+		AccountId: aws.String(d.Get("account_id").(string)),
+	}
+
+	var delegators []*organizations.DelegatedService
+	err := conn.ListDelegatedServicesForAccountPagesWithContext(ctx, input, func(page *organizations.ListDelegatedServicesForAccountOutput, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
+		}
+
+		delegators = page.DelegatedServices
```

Comment:
In case there are multiple pages returned, we should collect results over multiple pages

```suggestion
		delegators = append(delegators, page.DelegatedServices...)
```